### PR TITLE
SALTO-1042: support hidden instances

### PR DESCRIPTION
**Pull request by alonstern**
Added support for hidden annotation in element instances.

---
_Relese Notes:_ None

### CODE REVIEW

File: packages/workspace/src/workspace/hidden_values.ts [hunk 1]
```diff
@@ -341,8 +341,8 @@ const filterOutHiddenChanges = async (
       return change
     }
 
-    if (isType(baseElem) && isHidden(baseElem)) {
-      // A change of a hidden type should be omitted completely
+    if ((isType(baseElem) || isInstanceElement(baseElem)) && isHidden(baseElem)) {
```
Comment (netama) [inline packages/workspace/src/workspace/hidden_values.ts:344]: what's the use case? and do we have a test case for it? 
also - will it use `_hidden` directly on instances, or `_hidden_value` on the type?

Comment (alonstern) [reply packages/workspace/src/workspace/hidden_values.ts:344]: In my next PR, I will add a hidden instance which will contain internal information for the salesforce adapter
Added test for it
According to this PR there will be `_hidden` directly on the instance

**Comment by ori-moisis** [comment]
if we allow `_hidden` to be an annotation on instances, it needs to be defined [here](https://github.com/salto-io/salto/blob/master/packages/adapter-api/src/element_id.ts#L22:L22)

**Comment by ori-moisis** [comment]
we should not have two separate definitions for `_hidden` (CORE_ANNOTATIONS and INSTANCE_ANNOTATIONS) - there is nothing enforcing that they have the same value, but more importantly, if a reader tries to understand what `INSTANCE_ANNOTATIONS.HIDDEN` does they will not find anything because all the hidden value implementation uses CORE_ANNOTATIONS, so technically `INSTANCE_ANNOTATIONS.HIDDEN` doesn't have any meaning.

File: packages/adapter-api/src/element_id.ts [hunk 2]
```diff
@@ -23,6 +24,7 @@ export const INSTANCE_ANNOTATIONS = {
   DEPENDS_ON: '_depends_on',
   PARENT: '_parent',
   GENERATED_DEPENDENCIES: '_generated_dependencies',
+  HIDDEN: CORE_ANNOTATIONS.HIDDEN,
```
Comment (ori-moisis) [inline packages/adapter-api/src/element_id.ts:27]: nice trick
we should use it for the other instance annotations as well

Comment (ori-moisis) [reply packages/adapter-api/src/element_id.ts:27]: also, the new annotation should be added to `InstanceAnnotationTypes` as well

File: packages/adapter-api/src/builtins.ts [hunk 3]
```diff
@@ -53,6 +53,7 @@ export const InstanceAnnotationTypes: TypeMap = {
   [INSTANCE_ANNOTATIONS.DEPENDS_ON]: new ListType(StandardBuiltinTypes.STRING),
   [INSTANCE_ANNOTATIONS.PARENT]: new ListType(StandardBuiltinTypes.STRING),
   [INSTANCE_ANNOTATIONS.GENERATED_DEPENDENCIES]: new ListType(StandardBuiltinTypes.STRING),
+  [INSTANCE_ANNOTATIONS.HIDDEN]: StandardBuiltinTypes.BOOLEAN,
```
Comment (ori-moisis) [inline packages/adapter-api/src/builtins.ts:56]: nit: might be worth moving this or moving `INSTANCE_ANNOTATIONS` so they are defined next to each other (maybe move both to the new `core_annotations` file?), if we can't make typescript enforce them being related, we can at least put them next to each other

**Review by ori-moisis — APPROVED** [review]

File: packages/adapter-api/src/builtins.ts (hunk 3 above)
Comment (alonstern) [reply packages/adapter-api/src/builtins.ts:56]: Moving one next to the other creates a circular import, which forces us to move some more types around, so for now I am not sure this is worth it.

**Pull request merged**